private/bufpkg/bufmodule/uploader.go

```diff
@@ -39,37 +39,24 @@ type UploadOption func(*uploadOptions)
 // UploadWithLabels returns a new UploadOption that adds the given labels.
 //
 // This can be called multiple times. The unique result set of labels will be used.
-// We only ever allow one of labels, tags, or branchOrDraft set.
+// We only ever allow one of labels or tags to be set.
 func UploadWithLabels(labels ...string) UploadOption {
 	return func(uploadOptions *uploadOptions) {
 		uploadOptions.labels = append(uploadOptions.labels, labels...)
 	}
 }
 
 // UploadWithTags returns a new UploadOption that adds the given tags. This is handled
-// separately from labels because we disallow the use of the `--tag` flag when uploading a
-// workspace (e.g. a ModuleSet with 1+ target Modules).
+// separately from labels because we need to resolve the default label(s) when uploading.
 //
 // This can be called multiple times. The unique result set of tags will be used.
-// We only ever allow one of labels, tags, or branchOrDraft set.
+// We only ever allow one of labels or tags to be set.
 func UploadWithTags(tags ...string) UploadOption {
 	return func(uploadOptions *uploadOptions) {
 		uploadOptions.tags = append(uploadOptions.tags, tags...)
 	}
 }
 
-// UploadWithBranchOrDraft returns a new UploadOption that adds a branch/draft. This is
-// handled separately from labels because we disallow the use of `--branch`/`--draft` when
-// uploading a workspace (e.g. a ModuleSet with 1+ target Modules).
-//
-// If this is called multiple times, the last value is used.
-// We only ever allow one of labels, tags, or branchOrDraft set.
-func UploadWithBranchOrDraft(branchOrDraft string) UploadOption {
```

Comment:
Weren't we allowing branch and draft flags?

Reply:
We are -- since we are no longer gating branch/draft, then we no longer need to separate out branch/draft for upload, they are simply added as labels.